packages/common/src/models/Track.ts

```diff
@@ -89,6 +91,40 @@ export type PremiumConditions = {
   usdc_purchase?: PremiumConditionsUSDCPurchase
 }
 
+type PremiumConditionsCollectibleGated = {
+  type: PremiumContentType.COLLECTIBLE_GATED
+  nft_collection:
+    | PremiumConditionsEthNFTCollection
+    | PremiumConditionsSolNFTCollection
+}
+
+type PremiumConditionsFollowGated = {
+  type: PremiumContentType.FOLLOW_GATED
+  follow_user_id: number
+}
+
+type PremiumConditionsTipGated = {
+  type: PremiumContentType.TIP_GATED
+  tip_user_id: number
+}
+
+type PremiumConditionsUSDCPurchase = {
+  type: PremiumContentType.USDC_PURCHASE
+  usdc_purchase: {
+    price: StringUSDC
+    slot: number
+  }
+}
+
+export type PremiumConditionsWithType =
+  | PremiumConditionsCollectibleGated
+  | PremiumConditionsFollowGated
+  | PremiumConditionsTipGated
+  | PremiumConditionsUSDCPurchase
+
+export type PremiumConditions = PremiumConditionsBase &
```

Comment:
Does it not work if we do
```
type PremiumConditionsCollectibleGated = PremiumConditionsBase & {
  type: PremiumContentType.COLLECTIBLE_GATED
  nft_collection:
    | PremiumConditionsEthNFTCollection
    | PremiumConditionsSolNFTCollection
}
```
for each of the types and then just export a union of those types?
```
export type PremiumConditions =
  | PremiumConditionsCollectibleGated
  | PremiumConditionsFollowGated
  | PremiumConditionsTipGated
  | PremiumConditionsUSDCPurchase
```

My thinking was that it should never be possible to have a premium conditions object exist and _not_ have at least one of the conditions present. Seems like it would make some of the other logic easier.

Or is the issue that the response coming back from the server uses the same type and won't have the `type` field?

Reply:
think this works!